### PR TITLE
Wip/adrian web fixes

### PR DESCRIPTION
**Pull request by Akromx16**

### CODE REVIEW

File: gputop/remote/gputop-ui.js [hunk 1]
```diff
@@ -191,14 +165,19 @@ GputopUI.prototype.update_graphs = function(timestamp) {
...
-            counter.graph_data.push([mid, percent]);
-
+            if (counter.duration_dependent && counter.latest_duration) {
+                var per_sec_scale = 1000000000 / counter.latest_duration;
+                val *= per_sec_scale;
+                max *= per_sec_scale;
+                if (max > counter.inferred_max)
+                    counter.inferred_max = max;
+                if (val > counter.inferred_max)
+                    counter.inferred_max = val;
```
Comment (rib) [inline gputop/remote/gputop-ui.js]: We shouldn't be updating counter.inferred_max here and in gputop.js:append_counter_data()

We should update append_counter_data() to convert d_value and max into per-second units (if duration dependent) before updating counter.latest_value/latest_max/latest_duration/inferred_max.

gputop-ui.js:update_counter() should then also be updated to not repeat the conversion to per-second units.

**Comment by Akromx16** [comment]
The conversion to per second values is now done in gputop.js: append_counter, before we push the values to counter.updates. I also had to change the update_counter from gputop-ui so it doesn't do the conversion there anympore

File: gputop/remote/gputop.js [hunk 2]
```diff
@@ -131,27 +131,33 @@ function Counter () {
...
     this.test_mode = false;
 }
 
 Counter.prototype.append_counter_data = function (start_timestamp, end_timestamp,
                                                   d_value, max, reason) {
-     if (this.record_data && max != 0) {
-        this.updates.push([start_timestamp, end_timestamp, d_value, max, reason]);
 
+    if (this.duration_dependent && this.latest_duration) {
+        var per_sec_scale = 1000000000 / this.latest_duration;
```
Comment (rib) [inline gputop/remote/gputop.js]: At this point the duration we should use should be calculated from `end_timestamp - start_timestamp`

`this.latest_duration` is calculated in the same way later in this function (so the first time this function is called then `latest_duration` won't be initialized or will be out-of-date).

File: gputop/remote/gputop-ui.js [hunk 3]
```diff
@@ -255,17 +224,6 @@ GputopUI.prototype.update_counter = function(counter) {
         text_value *= 1000000;
     }
```
Comment (rib) [inline gputop/remote/gputop-ui.js]: I think we want to also make sure we convert mhz units to hz or us units to ns like this when plotting data in the trace view graphs.

If we introduced a `counter.units_scale` constant (default 1) then in `read_counter_xml` when we initialize `counter.duration_dependant` we could also check if (units === "mhz") { units = "hz"; counter.units_scale = 1000;} and if (units === "us") { units = "ns"; counter.units_scale = 1000; }

This in `append_counter_data` we can unconditionally do d_value *= counter.units_scale and max *= counter.units_scale.

Comment (rib) [inline gputop/remote/gputop-ui.js]: Er, I meant units.scale = 1000000; for mhz -> hz, oops

**Pull request merged**